codecparsers/jpegParser.cpp

```diff
@@ -313,6 +313,9 @@ bool Parser::nextMarker()
     const uint8_t* current = start;
     const uint8_t* match;
 
```

Comment:
@xuguangxin good point.

Unfortunately, @LindaYu17 says the coredump happens before this conditional when "end < current"... so this check would come too late.

@LindaYu17, does the issue only happen when m_size == 0?  That seems to be the only way "end < current" can occur (assuming currentBytePosition() is always < m_size).  If so, perhaps the following will fix it at the source of the problem:

```
const uint32_t epos = (m_size > 0 ? m_size - 1 : 0);
const uint8_t* const end = m_data + epos;
```

@LindaYu17 does Construct_InvalidParams unit test in jpegParser_unittest.cpp not detect your issue?  If not, can you write a unit test to catch this bug, please?